.gitignore

```diff
@@ -8,3 +8,29 @@ local.nix
 # Misc
 *.swp
 devices/*/kernel/config.*.old
+
+### env ###
+.env
+.envrc
+.direnv/
+
+### Vim ###
+# Swap
+[._]*.s[a-v][a-z]
+!*.svg 
+[._]*.sw[a-p]
+[._]s[a-rt-v][a-z]
+[._]ss[a-gi-z]
+[._]sw[a-p]
+
+# Session
+Session.vim
+Sessionx.vim
+
+# Temporary
+.netrwhist
+*~
+# Auto-generated tag files
+tags
+# Persistent undo
+[._]*.un~
```

Comment:
This'll be blunt, but "fix your editor" :).

I'm using vim too, and have it save its misc. files in a local directory to my user, not to the project.

Tough, really, it's also a great use of the global git ignore.

Yes, `*.swp` is in that list already. It shouldn't have been.

In a project's `.gitignore` file, only things the project "manages" should be present. E.g. here `.old` files from kernel normalization is fine; result files, image files.